include/zephyr/arch/arm/aarch32/cortex_m/scripts/linker.ld

```diff
@@ -232,6 +232,16 @@ SECTIONS
 	. = ALIGN(_region_min_align);
 	_image_ram_start = .;
 
+#if defined(CONFIG_SEGGER_RTT_SECTION_FIXED_ADDR)
+	SECTION_DATA_PROLOGUE(_RTT_SECTION_NAME,(NOLOAD),)
+	{
+	__rtt_buff_data_start = .;
+	*(.rtt_buff_data)
+	__rtt_buff_data_end = ALIGN(4);
+	} GROUP_DATA_LINK_IN(RAMABLE_REGION, RAMABLE_REGION)
+	__rtt_buff_data_size = __rtt_buff_data_end - __rtt_buff_data_start;
+#endif
+
```

Comment:
What about using `zephyr_linker_sources_ifdef()` instead?

Reply:
Isn't it a CMake function? Indeed I see it used only in CMakeLists.txt files, while linker file don't use it. Can it be used in linker files?

Reply:
No, it is used by CMake to embed linker script snippets.

Reply:
I've never done that way. On the other end, this linker script has similar sections with conditional compile switch, it was natural for me to do the same. Do you have an example?

Reply:
Would it be possible to separate this request in another PR?

Reply:
Here is how I did it when trying to implement the same as you have done here: https://github.com/sigvartmh/fw-nrfconnect-zephyr-1/commit/009b403829368e7a6687aa46a24d63f22d71a8b1#diff-7ab5276f0606a2bd1620a91984e170d371c28c4d7884774234ebb7340936ec06

Reply:
Sorry I forgot to include the segger_rtt.ld in my dump of this :/ it is pretty much the same as the section you have there.

Reply:
Thanks for the suggestions and sorry for my delay. It should be done now.